sched/distributor/poolDistributor_test.go

```diff
@@ -0,0 +1,68 @@
+package distributor
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+	"testing"
+)
+
+func TestPoolDistributor_NilCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nil)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_EmptyCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := make([]cm.Node, 0)
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nodes)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_OneNodeCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := cm.GenerateTestNodes(1)
+	clusterMock := cm.NewMockCluster(mockCtrl)
```

Comment:
I don't really understand the pushback on Mocks.  

I personally would rather use generated code instead of maintaining hand written fakes.  Its less work and less code for us to maintain.  If the interface changes I just run generate and a new mock is generated, vs having to go modify a bunch of hand written code that is untested.  

What drawbacks do you see to using mocks? @dbentley @jasonbs10